app/js/reducers/drugSearchReducer.js

```diff
@@ -1,35 +1,45 @@
-import * as actionTypes from '../actions/actionTypes';
+import {
+  SEARCH_DRUGS_LOADING,
+  SEARCH_DRUGS_FAILURE,
+  SEARCH_DRUGS_SUCCESS,
+  SELECT_DRUG,
+} from '../actions/actionTypes';
 
 const initialState = {
   drugs: [], selected: {}, error: null, loading: false,
 };
 
 export default (state = initialState, action) => {
   switch (action.type) {
-    case actionTypes.SEARCH_DRUGS_SUCCESS:
+    case SEARCH_DRUGS_SUCCESS:
       return {
         ...state,
         drugs: action.drugs,
-        selected: action.drugs && action.drugs.length === 1 ? action.drugs[0] : {},
+        selected: {},
         error: null,
```

Comment:
why are we setting error to null and loading to false, while they were initially null and false

Reply:
I will work on this and revert back. Thanks